src/components/text-input/TextInput.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import { Utils } from '../../shared';
+import Icon from '../../foundations/icon';
+
+import {
+    DEFAULT_TEXT_INPUT_SIZE,
+    TEXT_INPUT_VALUES_MAPPING,
+    DEFAULT_TEXT_INPUT_WIDTH,
+} from './TextInput.constants';
+import TextInputRoot from './TextInput.root';
+import InputRoot from './Input.root';
+import PTextInput from './TextInput.props';
+import LabelRoot from './Label.root';
+
+const TextInput: React.FC<PTextInput> = ({
+    label,
+    placeholder,
+    value,
+    leadingIcon,
+    trailingIcon,
+    backgroundColor,
+    size = DEFAULT_TEXT_INPUT_SIZE,
+    width = DEFAULT_TEXT_INPUT_WIDTH,
+    active = false,
+    hasError = false,
+    disabled = false,
+    animatedLabel = true,
+    onClear = Utils.noop,
+    onFocus = Utils.noop,
+    onChange = Utils.noop,
+    ...rest
+}: PTextInput) => {
+    const { iconSize } = TEXT_INPUT_VALUES_MAPPING[size];
+
+    const hasLabel = Utils.isString(label) && label.length > 0;
+    const onClearInput: () => void = Utils.isFunction(onClear) ? onClear : (): void => {};
```

Comment:
this can be removed, since we already assign `Utils.noop` to it in the props destructuring